File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/service/message/digitaltwin/GetSubmodelsRequest.java
package org.opcfoundation.webserver.service.message.digitaltwin;

import org.opcfoundation.webserver.types.digitaltwin.ObjectServiceContext;

public class GetSubmodelsRequest {
    private final String id;
    private final ObjectServiceContext context;

    public GetSubmodelsRequest(ObjectServiceContext context)
    {
        this.id = context.getObjectId().getId();
        this.context = context;
    }

    public String getId() {
        return id;
    }

    public ObjectServiceContext getContext() {
        return context;
    }
}

File: opcua-webservice-node/src/server/service/message/digitialtwin/GetSubmodelsRequest.ts
import { ObjectServiceContext } from "../../../types/digitaltwin/ObjectServiceContext";

export class GetSubmodelsRequest {
    private readonly _id: string;
    private readonly _context: ObjectServiceContext;

    constructor(context: ObjectServiceContext) {
        this._id = context.objectId.id;
        this._context = context;
    }

    get id(): string {
        return this._id;
    }

    get context(): ObjectServiceContext {
        return this._context;
    }
}